source/adapters/opencl/program.cpp

```diff
@@ -316,20 +316,54 @@ UR_APIEXPORT ur_result_t UR_APICALL urProgramSetSpecializationConstants(
   CL_RETURN_ON_FAILURE(clGetProgramInfo(CLProg, CL_PROGRAM_CONTEXT, sizeof(Ctx),
                                         &Ctx, &RetSize));
 
-  cl_ext::clSetProgramSpecializationConstant_fn F = nullptr;
-  const ur_result_t URResult = cl_ext::getExtFuncFromContext<decltype(F)>(
-      Ctx, cl_ext::ExtFuncPtrCache->clSetProgramSpecializationConstantCache,
-      cl_ext::SetProgramSpecializationConstantName, &F);
+  std::unique_ptr<std::vector<cl_device_id>> DevicesInCtx;
+  cl_adapter::getDevicesFromContext(cl_adapter::cast<ur_context_handle_t>(Ctx),
+                                    DevicesInCtx);
 
-  if (URResult != UR_RESULT_SUCCESS) {
-    return URResult;
-  }
+  cl_platform_id CurPlatform;
+  clGetDeviceInfo((*DevicesInCtx)[0], CL_DEVICE_PLATFORM,
+                  sizeof(cl_platform_id), &CurPlatform, nullptr);
+
+  oclv::OpenCLVersion PlatVer;
+  cl_adapter::getPlatformVersion(CurPlatform, PlatVer);
+
+  bool useExtensionLookup = false;
```

Comment:
`UseExtensionLookup` if we're sticking to the LLVM convention

Reply:
Fixed now, cheers.